gitlab.go

```diff
@@ -203,6 +204,18 @@ func NewClient(token string, options ...ClientOptionFunc) (*Client, error) {
 	return client, nil
 }
 
+// NewJobTokenClient returns a new GitLab API client. To use API methods which require
+// authentication, provide a valid job token(usually from CI_JOB_TOKEN environment variable).
+func NewJobTokenClient(token string, options ...ClientOptionFunc) (*Client, error) {
+	client, err := newClient(options...)
+	if err != nil {
+		return nil, err
+	}
+	client.authType = jobToken
+	client.token = token
+	return client, nil
+}
+
```

Comment:
Please order these alphabetically. In this case that means to order this function below the ``NewBasicAuthClient` function.